app/indexers/dataset_indexer.rb

```diff
@@ -13,7 +13,12 @@ class DatasetIndexer < Hyrax::WorkIndexer
   # Enable queries to distinguish top-level vs. nested datasets
   def generate_solr_document
    super.tap do |solr_doc|
-     solr_doc['top_level_bsi'] = object.in_works_ids.any? ? false : true
+     if object.in_works_ids.any?
+       solr_doc['top_level_bsi'] = false
```

Comment:
We can create a field constant for this index field in `Rdr::Index::Fields`:

```ruby
register :top_level, nil, solr_name: 'top_level_bsi'
```

Reply:
@dchandekstark I don't disagree with this but think it's outside the scope of this PR.  Plus, there are some other Solr fields for which we should probably do the same thing, so doing all those together might be a good task / PR.